File: problemas/1100/1118.py
opcao = 1

while opcao != 2:
    nota1 = float(input())

    while nota1 < 0 or nota1 > 10:
        print('nota invalida')
        nota1 = float(input())

    nota2 = float(input())

    while nota2 < 0 or nota2 > 10:
        print('nota invalida')
        nota2 = float(input())

    print(f'media = {(nota1 + nota2) / 2:.2f}')

    while True:
        print('novo calculo (1-sim 2-nao)')
        opcao = int(input())

        if opcao == 1 or opcao == 2:
            break

File: problemas/1100/1118.dart
import 'dart:io';

void main() {
  while (true) {
    double nota1 = double.parse(stdin.readLineSync());

    while (nota1 < 0 || nota1 > 10) {
      stdout.writeln('nota invalida');
      nota1 = double.parse(stdin.readLineSync());
    }

    double nota2 = double.parse(stdin.readLineSync());

    while (nota2 < 0 || nota2 > 10) {
      stdout.writeln('nota invalida');
      nota2 = double.parse(stdin.readLineSync());
    }

    stdout.writeln('media = ${((nota1 + nota2) / 2).toStringAsFixed(2)}');

    while (true) {
      stdout.writeln('novo calculo (1-sim 2-nao)');
      int opcao = int.parse(stdin.readLineSync());

      if (opcao == 1) {
        break;
      }

      if (opcao == 2) {
        return;
      }
    }
  }
}